node/src/command/solochain.rs

```diff
@@ -0,0 +1,419 @@
+                        shutdown_timeout.as_secs(),
+                    );
+                } else {
+                    log::error!(
+						"Task \"{}\" (Group: {}) was still running {}after waiting {} seconds to finish.",
+						task.name,
+						task.group,
+						instances_desc,
+						shutdown_timeout.as_secs(),
+					);
+                }
+            });
+        }
+
+        res.map_err(Into::into)
+    }
+}
+
+/// Equivalent to [Cli::create_runner]
+pub fn create_runner<T: CliConfiguration<DVC>, DVC: DefaultConfigurationValues>(
+    command: &T,
+) -> sc_cli::Result<SolochainRunner> {
+    let tokio_runtime = sc_cli::build_runtime()?;
+
+    // `capture` needs to be called in a tokio context.
+    // Also capture them as early as possible.
+    let signals = tokio_runtime.block_on(async { Signals::capture() })?;
+
+    init_cmd(command, &Cli::support_url(), &Cli::impl_version())?;
+
+    let base_path = command.base_path()?.unwrap();
+    let network_node_name = command.node_name()?;
+    let is_dev = command.is_dev()?;
+    let role = command.role(is_dev)?;
+    // TODO: where to get relay_chain_id from?
+    let relay_chain_id = "starlight_local_testnet".to_string();
+
+    let config = SolochainConfig {
+        tokio_handle: tokio_runtime.handle().clone(),
+        base_path,
+        network_node_name,
+        role,
+        relay_chain: relay_chain_id,
+    };
+
+    Ok(SolochainRunner {
+        config,
+        tokio_runtime,
+        signals,
+    })
+}
+
+/// The recommended open file descriptor limit to be configured for the process.
+const RECOMMENDED_OPEN_FILE_DESCRIPTOR_LIMIT: u64 = 10_000;
+
+/// Equivalent to [CliConfiguration::init]
+fn init_cmd<T: CliConfiguration<DVC>, DVC: DefaultConfigurationValues>(
+    this: &T,
+    support_url: &String,
+    impl_version: &String,
+) -> sc_cli::Result<()> {
+    sp_panic_handler::set(support_url, impl_version);
+
+    let mut logger = LoggerBuilder::new(this.log_filters()?);
+    logger
+        .with_log_reloading(this.enable_log_reloading()?)
+        .with_detailed_output(this.detailed_log_output()?);
+
+    if let Some(tracing_targets) = this.tracing_targets()? {
+        let tracing_receiver = this.tracing_receiver()?;
+        logger.with_profiling(tracing_receiver, tracing_targets);
+    }
+
+    if this.disable_log_color()? {
+        logger.with_colors(false);
+    }
+
+    logger.init()?;
+
+    match fdlimit::raise_fd_limit() {
+        Ok(fdlimit::Outcome::LimitRaised { to, .. }) => {
+            if to < RECOMMENDED_OPEN_FILE_DESCRIPTOR_LIMIT {
+                warn!(
+                    "Low open file descriptor limit configured for the process. \
+                        Current value: {:?}, recommended value: {:?}.",
+                    to, RECOMMENDED_OPEN_FILE_DESCRIPTOR_LIMIT,
+                );
+            }
+        }
+        Ok(fdlimit::Outcome::Unsupported) => {
+            // Unsupported platform (non-Linux)
```

Comment:
Should we be printing a warning here?

Reply:
This is copied from polkadot-sdk, so I'm not touching it

Reply:
*disregard*